scripts/examples/python/XenAPI/XenAPI.py

```diff
@@ -196,24 +206,31 @@
             self._session = result
             self.last_login_method = method
             self.last_login_params = params
-            self.API_version = self._get_api_version()
+
+            # This was initialized to 1.1 in the constructor.
+            # Now that we are logged in, the next time API_version() is run
+            # it can fetch the real version.
+            # However nothing in this script actually needs that, so don't call it immediately.
+            self._API_version = None
         except socket.error as e:
-            if e.errno == socket.errno.ETIMEDOUT:
+            # pytype false positive: there is a socket.errno in both py2 and py3
+            if e.errno == socket.errno.ETIMEDOUT: # pytype: disable=module-attr
```

Comment:
This was always like this, apparently some extra checks got enabled for python that affect feature branches.
I'll send a separate PR to fix it (since I can't push to this branch).

Reply:
`pylint` actually issues at lot more warnings on our files than it shows currently:

In order to not overwhelm use with too many messages, the `pylint` check uses `reviewdog` and configures `reviewdog` to only show the `pylint` warnings for the context of changed lines in the PR.

This helps to reduce the number of `pylint` warnings to those that are either directly caused by a PR or, if they pre-exist, are so close to the changed lines that the submitter should have the capacity to fix them with the PR.

The actions using `reviewdog` all have `continue-on-error: true`:

So even if `reviewdog` issued a comment, and even if the `step` fails for some unrelated reason, this should not be able to fail the CI.

It is a comment about the `pylint` message, which could even be ignored.

The intention would be to gracefully (over time) reduce the amount of `pylint` messages on files (or, at least, hopefully, prevent PRs causing more messages).

Reply:
Thanks, I've seen the CI fail, and I assumed it was due to the reviewdog comments (which are fixed by https://github.com/xapi-project/xen-api/pull/5761).
Looks like the failure is due to some Uri changes on master, I'll need to merge master to my branch again and fix the dependencies in dune on master.

Reply:
Fix here by merging from master again, and fixing the missing dependencies on master: https://github.com/xapi-project/xen-api/pull/5766

Reply:
Merged from master now.